app/validation/validator.py

```diff
@@ -329,27 +331,32 @@ def validate_skip_condition(self, skip_condition, answer_ids_with_group_id, bloc
 
     def validate_list_collector_type(self, block):  # noqa: C901  pylint: disable=too-complex
         errors = []
-        questions = []
+        add_questions = self._get_all_questions_for_block(block)
         add_answer_value = block['add_answer_value']
+        remove_answer_value = block['remove_answer_value']
 
-        if 'question' in block:
-            questions.append(block['question'])
-
-        if 'question_variants' in block:
-            for question in block['question_variants']:
-                questions.append(question)
-
-        for question in questions:
-            for answer in question['answers']:
+        for add_question in add_questions:
+            for answer in add_question['answers']:
                 if answer['type'] != 'Radio':
-                    errors.append(self._error_message('The list collector block {} does not contain a Radio answer type'
+                    errors.append(self._error_message('The list collector add block {} does not contain a Radio answer type'
                                                       .format(block['id'])))
 
                 if not self.options_contain_value(answer['options'], add_answer_value):
                     errors.append(
                         self._error_message('The list collector block {} has an add_answer_value that is not present in the answer values'
                                             .format(block['id'])))
 
+        for add_question in add_questions:
```

Comment:
How come this is duplicated?

Reply:
Good point, I had obviously copied but not modified this.